### ISSUE TITLE
Bug in FlushManager

### ISSUE THREAD

**Issue by xshaka** [Impetus/kundera#790]
Kundera FlushManager makes a static initialization of some map of this field:

```
    private static Map<EventType, List<CascadeType>> cascadePermission = new HashMap<EventType, List<CascadeType>>();
```

but the initialization violates Java object oriented design. See this:

```
    List<CascadeType> cascades = new ArrayList<CascadeType>();
    cascades.add(CascadeType.ALL);
    cascades.add(CascadeType.PERSIST);

    cascadePermission.put(EventType.INSERT, cascades);

    cascades.remove(CascadeType.PERSIST);
    cascades.add(CascadeType.MERGE);
    cascadePermission.put(EventType.UPDATE, cascades);

    cascades.remove(CascadeType.MERGE);
    cascades.add(CascadeType.REMOVE);
    cascadePermission.put(EventType.DELETE, cascades);
```

the idea behind the code is to have the following mapping:

```
    EventType.INSERT, {CascadeType.ALL, CascadeType.PERSIST}
    EventType.UPDATE, {CascadeType.ALL, CascadeType.MERGE}
    EventType.DELETE, {CascadeType.ALL, CascadeType.REMOVE}
```

but the result it gets is having:

```
    EventType.INSERT, {CascadeType.ALL, CascadeType.REMOVE}
    EventType.UPDATE, {CascadeType.ALL, CascadeType.REMOVE}
    EventType.DELETE, {CascadeType.ALL, CascadeType.REMOVE}
```

because what every single entry is pointing to is a reference, and that reference is changed over and over again. The behavior we get is that something like: 

```
    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE}) 
```

simply doesn't work.

**Comment by karthikprasad13** [comment]
@xshaka 

Thank you for your contribution in helping us fixing the bugs. The fix has been updated in current trunk and will be available with next release

-Karthik

**Comment by chhavigangwal** [comment]
Fix released with Kundera-3.2.
